tests/kernel/testHyperRectDomain.cpp

```diff
@@ -308,88 +368,165 @@ bool testSTLCompat()
+  range = domain.subRange( 1, domain.lowerBound()  );
+  REQUIRE( range.begin() == range.end() );
+  REQUIRE( range.rbegin() == range.rend() );
+}
 
-  range = domain.subRange( 2, 1, 0, domain.lowerBound()  );
-  ++nb; nbok += range.begin() == range.end() ? 1 : 0;
-  trace.info() << "(" << nbok << "/" << nb << ") Checking that begin() == end() for sub-range {2,1,0}" << std::endl;
+///////////////////////////////////////////////////////////////////////////////
+// Benchmarks
+//
+// Note: using multiple TEST_CASE instead of SECTION because it seems to alter
+// the way the compiler optimizes the code and it reduces performances.
 
-  ++nb; nbok += range.rbegin() == range.rend() ? 1 : 0;
-  trace.info() << "(" << nbok << "/" << nb << ") Checking that rbegin() == rend() for sub-range {2,1,0}" << std::endl;
+/// Timer user in tic and toc
+auto tic_timer = std::chrono::high_resolution_clock::now();
 
-  range = domain.subRange( 0, 2, domain.lowerBound()  );
-  ++nb; nbok += range.begin() == range.end() ? 1 : 0;
-  trace.info() << "(" << nbok << "/" << nb << ") Checking that begin() == end() for sub-range {0,2}" << std::endl;
+/// Starts timer
+void tic()
+{
+  tic_timer = std::chrono::high_resolution_clock::now();
+}
 
-  ++nb; nbok += range.rbegin() == range.rend() ? 1 : 0;
-  trace.info() << "(" << nbok << "/" << nb << ") Checking that rbegin() == rend() for sub-range {0,2}" << std::endl;
+/// Ends timer and return elapsed time
+double toc()
+{
+  const auto toc_timer = std::chrono::high_resolution_clock::now();
+  const std::chrono::duration<double> time_span = toc_timer - tic_timer;
+  return time_span.count();
+}
 
-  range = domain.subRange( 2, 0, domain.lowerBound()  );
-  ++nb; nbok += range.begin() == range.end() ? 1 : 0;
-  trace.info() << "(" << nbok << "/" << nb << ") Checking that begin() == end() for sub-range {2,0}" << std::endl;
+// Context for each benchmark
+struct BenchDomain
+{
+  static constexpr std::size_t dim = 3;
+  static constexpr signed long long int size = 1000;
 
-  ++nb; nbok += range.rbegin() == range.rend() ? 1 : 0;
-  trace.info() << "(" << nbok << "/" << nb << ") Checking that rbegin() == rend() for sub-range {2,0}" << std::endl;
+  using Space = DGtal::SpaceND<dim>;
+  using Point = Space::Point;
+  using Domain = DGtal::HyperRectDomain<Space>;
 
-  range = domain.subRange( 1, domain.lowerBound()  );
-  ++nb; nbok += range.begin() == range.end() ? 1 : 0;
-  trace.info() << "(" << nbok << "/" << nb << ") Checking that begin() == end() for sub-range {1}" << std::endl;
+  BenchDomain()
+    : a(Point::diagonal(0))
+    , b(Point::diagonal(size))
+    , domain(Domain(a, b))
+    {
+    }
 
-  ++nb; nbok += range.rbegin() == range.rend() ? 1 : 0;
-  trace.info() << "(" << nbok << "/" << nb << ") Checking that rbegin() == rend() for sub-rang {1}" << std::endl;
+  Point a, b;
+  Domain domain;
+};
 
-  trace.endBlock();
-  return nb == nbok;
+TEST_CASE_METHOD( BenchDomain, "Benchmarking domain traversal", "[.bench]" )
```

Comment:
I don't understand your benchmark framework. Haven't used the Catch2 micro-benchmarking for that?

I don't see these test cases in the `./testHyperRectDomain -l`. Is that normal ?

Reply:
The dot at the beginning of the tag makes it hidden, ie it is not executed during the normal test run. However, I think the tag was listed with the `-t` option but it is not the case (maybe since the new version of Catch?). Anyway, you can run it with the `[bench]` option...

Reply:
I’ll try, thx

Reply:
Indeed `[bench]` is hidden when doing a `-t`. what do you mean by "Anyway, you can run it with the [bench] option..." ?

Reply:
(and what is the diff between `TEST_CASE` and `TEST_CASE_METHOD`? )

Reply:
> Indeed `[bench]` is hidden when doing a `-t`. what do you mean by "Anyway, you can run it with the [bench] option..." ?

The "anyway" was maybe not appropriate :sweat_smile: I mean, you can call this hidden tag this way:
```bash
./test/kernel/testHyperRectDomain [bench]
```
```

Reply:
> (and what is the diff between `TEST_CASE` and `TEST_CASE_METHOD`? )
> 
> (no doc in the catch2 repo https://github.com/catchorg/Catch2/blob/master/docs/test-cases-and-sections.md)

It is explained here: https://github.com/catchorg/Catch2/blob/master/docs/test-fixtures.md
I used it to launch the test as a class method, thus avoiding to repeat the context creation (typedefs and domain definition).

Reply:
```
./testHyperRectDomain [bench]
zsh: no matches found: [bench]
```

Reply:
```
./testHyperRectDomain -t bench
Tags for matching test cases:
0 tags
```

Reply:
And with quotes around `[bench]`? It seems to be interpreted by your shell instead of being passed to the binary...

Reply:
working with the double-quotes..

Reply:
thx

Reply:
> I don't understand your benchmark framework. Haven't used the Catch2 micro-benchmarking for that?

And about the included micro-benchmark, I have made some tests before the summer and I had some inconsistent results that I was trying to understand. 

One other strange thing is the performance of `subRange` iterator compared to the "full" iterator...